<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Stanislav Gabri≈°</author>
 <class>EditorWindow</class>
 <widget class="QMainWindow" name="EditorWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>965</width>
    <height>736</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <spacer name="verticalSpacer_3">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QWidget" name="widget_4" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>771</width>
        <height>239</height>
       </size>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QWidget" name="widget" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>271</width>
           <height>221</height>
          </size>
         </property>
         <widget class="TextEditNewline" name="textEdit">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>40</y>
            <width>271</width>
            <height>161</height>
           </rect>
          </property>
          <property name="toolTip">
           <string>Input format: [anything] [type] [name], [pin name]
where type, name and comma are compulsory.
Number of spaces is unimportant.
e.g.: 
1:   int a, IN
int f,A</string>
          </property>
          <property name="toolTipDuration">
           <number>8000</number>
          </property>
          <property name="styleSheet">
           <string notr="true">QTextEdit{font: 15px}
QToolTip{
font:12px
}</string>
          </property>
          <property name="autoFormatting">
           <set>QTextEdit::AutoNone</set>
          </property>
          <property name="lineWrapMode">
           <enum>QTextEdit::NoWrap</enum>
          </property>
         </widget>
         <widget class="QLabel" name="label">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>9</y>
            <width>81</width>
            <height>31</height>
           </rect>
          </property>
          <property name="styleSheet">
           <string notr="true">font: 15px</string>
          </property>
          <property name="text">
           <string>Inputs:</string>
          </property>
          <property name="scaledContents">
           <bool>false</bool>
          </property>
         </widget>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QFrame" name="frame">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>191</width>
           <height>221</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="frameShape">
          <enum>QFrame::Box</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <property name="lineWidth">
          <number>1</number>
         </property>
         <widget class="QPushButton" name="pushButton">
          <property name="geometry">
           <rect>
            <x>130</x>
            <y>70</y>
            <width>25</width>
            <height>22</height>
           </rect>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Add new category</string>
          </property>
          <property name="toolTipDuration">
           <number>3000</number>
          </property>
          <property name="text">
           <string>+</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_5">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>120</y>
            <width>151</width>
            <height>91</height>
           </rect>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="pixmap">
           <pixmap resource="Resources.qrc">:/icons/icons/generic_block.png</pixmap>
          </property>
          <property name="scaledContents">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QLineEdit" name="lineEdit">
          <property name="geometry">
           <rect>
            <x>40</x>
            <y>20</y>
            <width>113</width>
            <height>21</height>
           </rect>
          </property>
          <property name="placeholderText">
           <string>Block name</string>
          </property>
         </widget>
         <widget class="QComboBox" name="comboBox">
          <property name="geometry">
           <rect>
            <x>40</x>
            <y>70</y>
            <width>81</width>
            <height>22</height>
           </rect>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="editable">
           <bool>false</bool>
          </property>
          <property name="currentText">
           <string/>
          </property>
          <property name="insertPolicy">
           <enum>QComboBox::InsertAtTop</enum>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QComboBox::AdjustToContentsOnFirstShow</enum>
          </property>
          <property name="duplicatesEnabled">
           <bool>false</bool>
          </property>
         </widget>
         <widget class="QLabel" name="label_4">
          <property name="geometry">
           <rect>
            <x>40</x>
            <y>50</y>
            <width>111</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>Category</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
         <widget class="QLabel" name="label_6">
          <property name="geometry">
           <rect>
            <x>40</x>
            <y>100</y>
            <width>111</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>Pin numbering</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
         <widget class="QLabel" name="label_3">
          <property name="geometry">
           <rect>
            <x>40</x>
            <y>0</y>
            <width>111</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Name of Block</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QWidget" name="widget_2" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>271</width>
           <height>221</height>
          </size>
         </property>
         <widget class="TextEditNewline" name="textEdit_2">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>40</y>
            <width>271</width>
            <height>161</height>
           </rect>
          </property>
          <property name="toolTip">
           <string>Output format: [type] [name], [pin name]
where type, name and comma are compulsory.
Number of spaces is unimportant.
e.g.:
int a, OUT
2:  int k,  R</string>
          </property>
          <property name="toolTipDuration">
           <number>8000</number>
          </property>
          <property name="styleSheet">
           <string notr="true">QTextEdit{font: 15px}
QToolTip{
font:12px
}</string>
          </property>
          <property name="lineWrapMode">
           <enum>QTextEdit::NoWrap</enum>
          </property>
         </widget>
         <widget class="QLabel" name="label_2">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>10</y>
            <width>81</width>
            <height>31</height>
           </rect>
          </property>
          <property name="styleSheet">
           <string notr="true">font: 15px</string>
          </property>
          <property name="text">
           <string>Outputs:</string>
          </property>
          <property name="scaledContents">
           <bool>false</bool>
          </property>
         </widget>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QWidget" name="widget_5" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <spacer name="horizontalSpacer_5">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QWidget" name="widget_3" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>818</width>
           <height>265</height>
          </size>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="label_7">
            <property name="styleSheet">
             <string notr="true">font: 15px</string>
            </property>
            <property name="text">
             <string>Block logic (do not declare inputs and outputs again!(use them without prefixing a type)):</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="textEdit_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>1</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>800</width>
              <height>231</height>
             </size>
            </property>
            <property name="lineWrapMode">
             <enum>QTextEdit::NoWrap</enum>
            </property>
            <property name="placeholderText">
             <string>Enter your code here</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_6">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>965</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuNew">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
   </widget>
   <widget class="QMenu" name="menuNew_2">
    <property name="title">
     <string>New</string>
    </property>
    <addaction name="actionBlank_Block"/>
   </widget>
   <widget class="QMenu" name="menuGeneral">
    <property name="title">
     <string>General</string>
    </property>
    <addaction name="actionHelp"/>
   </widget>
   <addaction name="menuNew"/>
   <addaction name="menuNew_2"/>
   <addaction name="menuGeneral"/>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="statusTip">
    <string/>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="Resources.qrc">
     <normaloff>:/icons/icons/open.png</normaloff>:/icons/icons/open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionBlank_Block">
   <property name="icon">
    <iconset resource="Resources.qrc">
     <normaloff>:/icons/icons/new.png</normaloff>:/icons/icons/new.png</iconset>
   </property>
   <property name="text">
    <string>Blank Block</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="Resources.qrc">
     <normaloff>:/icons/icons/save.png</normaloff>:/icons/icons/save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="icon">
    <iconset resource="Resources.qrc">
     <normaloff>:/icons/icons/help.png</normaloff>:/icons/icons/help.png</iconset>
   </property>
   <property name="text">
    <string>Help</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TextEditNewline</class>
   <extends>QTextEdit</extends>
   <header>texteditnewline.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="Resources.qrc"/>
 </resources>
 <connections/>
</ui>
